internal/hmyapi/apiv1/blockchain.go

```diff
@@ -66,7 +66,9 @@ func (s *PublicBlockChainAPI) isBeaconShard() error {
 }
 
 func (s *PublicBlockChainAPI) isBlockGreaterThanLatest(blockNum rpc.BlockNumber) error {
-	if uint64(blockNum) > s.b.CurrentBlock().NumberU64() {
+	if blockNum != rpc.PendingBlockNumber &&
```

Comment:
Can you add a comment on why you need to do these two extra comparison?

Reply:
pretty obvious because rpc.BlockNumber in int64 (latest = -1. pending = -2) and currentblock is uint64. and casting negative int64 to uint64 cast to large value for two's compliment? but okay.